src/index.ts

```diff
@@ -1,17 +1,47 @@
 import * as core from "@actions/core";
 import { readConfig } from "./config";
 import { apiClient } from "./client/apiClient";
+import { isSuccessful } from "./run/status";
+import { startRun } from "./run/start";
+import { FinishedRun, waitForRunEnd } from "./run/ongoing";
 
 async function run(): Promise<void> {
   try {
     const config = readConfig();
     const client = apiClient(config.api);
-    const simulations = await client.getSimulations(); // TODO RND-6 Start simulation instead
-    core.setOutput("result", `Found {} ${simulations.length} simulations.`);
+
+    await client.checkCloudCompatibility();
+
+    const startedRun = await startRun(client, config);
+    core.setOutput("run_id", startedRun.runId);
+    core.setOutput("reports_url", startedRun.reportsUrl);
+    core.setOutput("runs_url", startedRun.runsUrl);
+
+    const finishedRun = await waitForRunEnd(client, startedRun);
+    logResult(finishedRun);
+
+    // TODO RND-7 Metrics summary and XML report
+
+    core.setOutput("runs_status_code", finishedRun.statusCode);
+    core.setOutput("runs_status_name", finishedRun.statusName);
+    core.setOutput("run_assertions", JSON.stringify(finishedRun.assertions));
   } catch (error) {
     const msg = error instanceof Error ? error.message : `Unknown error: ${error}`;
     core.setFailed(msg);
   }
 }
 
+const logResult = (finishedRun: FinishedRun) => {
+  if (!isSuccessful(finishedRun.statusCode)) {
+    core.setFailed(`Run ${finishedRun.runId} failed with status ${finishedRun.statusName}`);
+  } else {
+    core.info(`Run ${finishedRun.runId} finished with status ${finishedRun.statusName}`);
+  }
+
+  for (const assertion of finishedRun.assertions) {
```

Comment:
Thanks!

I didn't know about `for ... of` construction.
I learned a thing today :tada: